docs/source/aggregation.rst

```diff
@@ -0,0 +1,184 @@
+
+- *All jobs*: All of the project's jobs are passed to the operation function.
+- *Group by state point key(s)*: The aggregates are grouped by one or more state point keys.
+- *Group by arbitrary key function*: The aggregates are grouped by keys determined by a key-function that expects an instance of :class:`~.signac.contrib.job.Job` and return the grouping key.
+- Grouping into aggregates of a specific size.
+- Using a completely custom aggregator function when even greater flexibility is needed.
+
+Group By
+--------
+
+:class:`~flow.aggregator.groupby` allows users to aggregate jobs by grouping them by a state point key, an iterable of state point keys whose values define the groupings, or an arbitrary callable of :class:`~signac.contrib.job.Job`.
+
+.. code-block:: python
+
+    @aggregator.groupby("temperature")
+    @Project.operation
+    def op3(*jobs):
+        pass
+
+In the above example, the jobs will be aggregated based on the state point key ``"temperature"``.
+So, all the jobs having the same value of **temperature** in their state point will be aggregated together.
+
+Groups Of
+---------
+
+:class:`~flow.aggregator.groupsof` allows users to aggregate jobs by generating aggregates of a given size.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2)
+    @Project.operation
+    def op4(job1, job2=None):
+        pass
+
+In the above example, the jobs will get aggregated in groups of 2 and hence, up to two jobs will be passed as arguments at once.
+
+.. note::
+
+    In case the number of jobs in the project in this example is odd, there will be one aggregate containing only a single job.
+    In general, the last aggregate from :class:`~flow.aggregator.groupsof` will contain the remaining jobs if the aggregate size does not evenly divide the number of jobs in the project.
+    If a remainder is expected and valid, users should make sure that the operation function can be called with the reduced number of arguments (e.g. by using ``*jobs`` or providing default arguments as shown above).
+
+Sorting jobs for aggregation
+----------------------------
+
+Aggregators allow users to sort the jobs before creating aggregates with the ``sort_by`` parameter.
+The sorting order can be defined with the ``sort_ascending`` parameter.
+By default, when no ``sort_by`` parameter is specified, the order of the jobs will be decided by the iteration order of the **signac** project.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2, sort_by='temperature', sort_ascending=False)
+    @Project.operation
+    def op5(*jobs):
+        pass
+
+.. note::
+
+    In the above example, all the jobs will be sorted by the state point parameter ``"temperature"`` in descending order and then be aggregated as groups of 2.
+
+Selecting jobs for aggregation
+------------------------------
+
+**signac-flow** allows users to selectively choose which jobs to pass into operation functions.
```

Comment:
```suggestion
**signac-flow** allows users to selectively choose which jobs to pass into operation functions.
This can be used to generate aggregates from only the selected jobs, excluding any jobs that do not meet the selection criteria.
```